File: LifeAgent.Api/Models/LifeReview.cs
namespace LifeAgent.Api.Models;

public class LifeReviewRequest
{
    public string? ClientTimeZone { get; set; }
    public int? Limit { get; set; }
    public string? Period { get; set; }
}

public class LifeReviewResponse
{
    public bool Success { get; set; } = true;
    public string Period { get; set; } = "recent";
    public string WindowLabel { get; set; } = "最近";
    public IReadOnlyList<LifeReviewCard> Cards { get; set; } = Array.Empty<LifeReviewCard>();
    public IReadOnlyList<LifeReviewTheme> ReviewThemes { get; set; } = Array.Empty<LifeReviewTheme>();
    public IReadOnlyList<LifeReviewContinuityHint> ContinuityHints { get; set; } = Array.Empty<LifeReviewContinuityHint>();
    public IReadOnlyList<LifeReviewSourceEvent> SourceEvents { get; set; } = Array.Empty<LifeReviewSourceEvent>();
    public int UsedEventCount { get; set; }
    public int UsedMemoryCount { get; set; }
    public int UsedPlanSignalCount { get; set; }
    public bool ReadOnly { get; set; } = true;
    public bool WroteData { get; set; } = false;
    public bool Executed { get; set; } = false;
}

public class LifeReviewCard
{
    public string Id { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Text { get; set; } = string.Empty;
    public IReadOnlyList<string> SourceEventIds { get; set; } = Array.Empty<string>();
    public IReadOnlyList<LifeReviewEvidenceHint> EvidenceHints { get; set; } = Array.Empty<LifeReviewEvidenceHint>();
}

public class LifeReviewTheme
{
    public string Id { get; set; } = string.Empty;
    public string Kind { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Summary { get; set; } = string.Empty;
    public string Href { get; set; } = string.Empty;
    public string ActionLabel { get; set; } = string.Empty;
    public IReadOnlyList<LifeReviewEvidenceHint> EvidenceHints { get; set; } = Array.Empty<LifeReviewEvidenceHint>();
}

public class LifeReviewContinuityHint
{
    public string Id { get; set; } = string.Empty;
    public string Kind { get; set; } = string.Empty;
    public string Label { get; set; } = string.Empty;
    public string Detail { get; set; } = string.Empty;
    public string Href { get; set; } = string.Empty;
    public string Reason { get; set; } = string.Empty;
}

public class LifeReviewEvidenceHint
{
    public string Kind { get; set; } = string.Empty;
    public string Label { get; set; } = string.Empty;
    public string Text { get; set; } = string.Empty;
    public string Reason { get; set; } = string.Empty;
    public string Href { get; set; } = string.Empty;
}

public class LifeReviewSourceEvent
{
    public string Id { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public string OccurredAt { get; set; } = string.Empty;
}

public class LifeReviewKeepRequest
{
    public string CardId { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Text { get; set; } = string.Empty;
    public IReadOnlyList<string> SourceEventIds { get; set; } = Array.Empty<string>();
}

File: life-agent-web/src/app/actions/lifeReview.ts
"use server";

import { getToken } from "./auth";

const API_BASE = process.env.API_BASE_URL || "http://localhost:5140";

export interface LifeReviewCard {
  id: string;
  title: string;
  text: string;
  sourceEventIds: string[];
}

export interface LifeReviewSourceEvent {
  id: string;
  title: string;
  content: string;
  occurredAt: string;
}

export interface LifeReviewResponse {
  success: boolean;
  period: LifeReviewPeriod;
  windowLabel: string;
  cards: LifeReviewCard[];
  sourceEvents: LifeReviewSourceEvent[];
  usedEventCount: number;
  usedMemoryCount: number;
  usedPlanSignalCount: number;
  readOnly: boolean;
  wroteData: boolean;
  executed: boolean;
}

export type LifeReviewPeriod = "recent" | "today" | "week";

export interface KeepLifeReviewCardResponse {
  success: boolean;
  previewOnly: boolean;
  memoryWriteEnabled: boolean;
  wroteMemory: boolean;
  wroteReviewState: boolean;
  memoryId?: string | null;
  data: {
    id: string;
    title: string;
    reviewStatus?: "pending" | "kept" | "dismissed" | "remembered";
  };
}

export async function getLifeReview(
  clientTimeZone?: string,
  limit = 30,
  period: LifeReviewPeriod = "recent"
): Promise<LifeReviewResponse> {
  const token = await getToken();
  if (!token) throw new Error("Unauthorized");

  const res = await fetch(`${API_BASE}/api/life/review`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      Authorization: `Bearer ${token}`,
    },
    body: JSON.stringify({
      clientTimeZone: clientTimeZone || "Asia/Shanghai",
      limit,
      period,
    }),
    cache: "no-store",
  });

  const data = await res.json();
  if (!res.ok) {
    throw new Error(data.error?.message || "暂时无法整理最近回顾");
  }

  return data;
}

export async function keepLifeReviewCard(
  card: LifeReviewCard
): Promise<KeepLifeReviewCardResponse> {
  const token = await getToken();
  if (!token) throw new Error("Unauthorized");

  const res = await fetch(`${API_BASE}/api/life/review/cards/keep`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      Authorization: `Bearer ${token}`,
    },
    body: JSON.stringify({
      cardId: card.id,
      title: card.title,
      text: card.text,
      sourceEventIds: card.sourceEventIds,
    }),
    cache: "no-store",
  });

  const data = await res.json();
  if (!res.ok) {
    throw new Error(data.error?.message || "暂时不能放入记忆候选");
  }

  return data;
}
